src/parser.rs

```diff
@@ -1031,7 +1031,7 @@ impl<'a> Parser<'a> {
                 }
             }
             None => return Ok((None, None)),
-            _ => panic!("Programming error. parse_query_and_fragment() called without ? or # {:?}")
```

Comment:
This compiles because the `panic!` macro has a special case for use with a single "argument" that by-passes the string formatting system entirely:

https://github.com/rust-lang/rust/blob/a0da1e0653/src/libstd/macros.rs#L43-L49